python/runtime/cudaq/algorithms/py_run.cpp

```diff
@@ -0,0 +1,168 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+#include "cudaq/algorithms/run.h"
+#include "utils/OpaqueArguments.h"
+#include "mlir/Bindings/Python/PybindAdaptors.h"
+#include <pybind11/complex.h>
+#include <pybind11/stl.h>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include <iostream>
+
+namespace cudaq {
+
+void pyAltLaunchKernel(const std::string &, MlirModule, OpaqueArguments &,
+                       const std::vector<std::string> &);
```

Comment:
This decl ought to be moved to a header so we don't end up with a lot of copies of it.

Reply:
done, thanks!